docs/metrics.md

```diff
@@ -4,30 +4,189 @@ Let's look at the various metrics you can calculate using Valor.
-| Mean IOU 	| The average of IOUs, calculated over several different classes. 	|
+| Name | Description | Equation |
+| :- | :- | :- |
+| Intersection Over Union (IoU) | A ratio between the groundtruth and predicted regions of an image, measured as a percentage, grouped by class. |$$\dfrac{area( prediction \cap groundtruth )}{area( prediction \cup groundtruth )}$$ |
+| Mean IoU 	| The average of IoUs, calculated over several different classes. | $$\dfrac{1}{\|classes\|} \sum\limits_{c \in classes} IoU_{c}$$ |
+
+# Notes
+1. When calculating IoUs for object detection metrics, Valor handles the necessary conversion between different types of geometric annotations. For example, if your model prediction is a polygon and your groundtruth is a raster, then the raster will be converted to a polygon prior to calculating the IoU.
+
+# Appendix: Metric Calculations
+
+## Binary ROC AUC
+
+### Determining the rate of correct predictions.
+
+| Element | Description |
+| ------- | ------------ |
+| True Positive (TP) | Prediction confidence score >= threshold and is correct. |
+| False Positive (FP) | Prediction confidence score >= threshold and is incorrect. |
+| True Negative (TN) | Prediction confidence score < threshold and is correct. |
+| False Negative (FN) | Prediction confidence score < threshold and is incorrect. |
+
+- $\text{True Positive Rate (TPR)} = \dfrac{|TP|}{|TP| + |FN|}$
+
+- $\text{False Positive Rate (FPR)} = \dfrac{|FP|}{|FP| + |TN|}$
+
+These rates vary depending on a threshold so we can add a threshold parameter to be more explicit.
+
+$$
+TPR(threshold) = \dfrac{|TP(threshold)|}{|TP(threshold)| + |FN(threshold)|}
+$$
+
+$$
+FPR(threshold) = \dfrac{|FP(threshold)|}{|FP(threshold)| + |TN(threshold)|}
+$$
+
+### Receiver Operating Characteristic (ROC)
+
+An ROC curve plots TPR vs. FPR at different confidence thresholds.
+
+In Valor, we use the confidence scores sorted in decreasing order as our thresholds.
+
+$$
+Point(score) = (FPR(score), \ TPR(score))
+$$
+
+### Area under the ROC curve (ROC AUC)
+
+From the ROC curve we calculate the ROC AUC metric by approximating the integral using the trapezoidal rule formula.
+
+$$
+ROC AUC =  \sum_{i=1}^{|scores|} \frac{  \lVert Point(score_{i-1}) - Point(score_i) \rVert }{2}
+$$
+
+### References
+- [Classification: ROC Curve and AUC](https://developers.google.com/machine-learning/crash-course/classification/roc-and-auc)
+
+## Average Precision (AP)
+
+For object-detection and instance segmentation tasks, average-precision is calculated from the intersection-over-union (IoU) of geometric annotations.
```

Comment:
i prefer replacing "annotations" with "predictions and groundtruths". 

people usually don't think of an "annotation" as a prediction